sisl/viz/plotly/plots/bands.py

```diff
@@ -0,0 +1,693 @@
+from functools import partial
+import itertools
+
+import numpy as np
+import xarray as xr
+import plotly.express as px
+
+import sisl
+from ..plot import Plot, PLOTS_CONSTANTS, entry_point
+from ..plotutils import find_files
+from ..input_fields import TextInput, SwitchInput, ColorPicker, DropdownInput,\
+     IntegerInput, FloatInput, RangeInput, RangeSlider, QueriesInput, ProgramaticInput, FunctionInput, SileInput, \
+         PlotableInput, SpinSelect
+from ..input_fields.range import ErangeInput
+
+
+class BandsPlot(Plot):
+    """
+    Plot representation of the bands.
+
+    Parameters
+    -------------
+    bands_file: bandsSileSiesta, optional
+        This parameter explicitly sets a .bands file. Otherwise, the bands
+        file is attempted to read from the fdf file
+    band_structure: BandStructure, optional
+        The BandStructure object to be used.
+    add_band_trace_data:  optional
+        A function that receives each band (as a DataArray) and adds data to
+        the trace. It also recieves the plot object.              The
+        returned data may even overwrite the existing one, therefore it can
+        be useful to fully customize your bands plot (individual style for
+        each band if you want).
+    eigenstate_map:  optional
+        This function receives the eigenstate object for each k value when
+        the bands are being extracted from a hamiltonian.             You can
+        do whatever you want with it, the point of this function is to avoid
+        running the diagonalization process twice.
+    Erange: array-like of shape (2,), optional
+        Energy range where the bands are displayed.
+    E0: float, optional
+        The energy to which all energies will be referenced (including
+        Erange).
+    bands_range: array-like of shape (2,), optional
+        The bands that should be displayed. Only relevant if Erange is None.
+    path: array-like of dict, optional
+        Path along which bands are drawn in units of reciprocal lattice
+        vectors.             Note that if you want to provide a path
+        programatically you can do it more easily with the `band_structure`
+        setting
+    spin:  optional
+        Determines how the different spin configurations should be displayed.
+        In spin polarized calculations, it allows you to choose between spin
+        0 and 1.             In non-colinear spin calculations, it allows you
+        to ask for a given spin texture,             by specifying the
+        direction.
+    spin_texture_colorscale: str, optional
+        The plotly colorscale to use for the spin texture (if displayed)
+    gap: bool, optional
+        Whether the gap should be displayed in the plot
```

Comment:
Perhaps also Gamma-Gamma gap?

Reply:
Is it possible that there is no gamma in the band structure? :sweat_smile: 

What about `custom_gaps` so that the user can do `custom_gaps=[("Gamma", "Gamma"), ("Gamma", "X"), ("X", 0.456)]`?

Reply:
Yeah, it was more that in `Demo.ipynb` 2 lines as I recall, but not Gamma-Gamma.

But yeah, custom gaps would be nice (or allow gap be a `bool` or `tuple of str`)

Reply:
> or allow gap be a bool or tuple of str

This would be the way to go in python, but I need to implement an input field that supports multiple input types for it to work in the GUI. It is something that I wanted to do, but I have not done yet.

Reply:
Ok, great.

Reply:
With the last commit, you can do:

```python
bands_plot.update_settings(custom_gaps=[
    {"from": "Gamma", "to": "X", "color": "green"},
    {"from": "Gamma", "to": 0.14}
])
```

to get the gaps that you want. When input fields that allow multiple data types (with nested input fields) are implemented, I will merge it to the `gap` parameter. Also I have plans that `QueriesInput`, which is what I'm using for `custom_gaps` doesn't care whether a dict or a tuple is passed (makes sense since keys are ordered anyways to display them in the GUI). In that way, `[("Gamma", "X", "green")]` will be automatically parsed by the input field to `[{"from": "Gamma", "to": "X", "color": "green"}]` :)